content-src/components/NewTabPage/NewTabPage.js

```diff
@@ -42,11 +43,31 @@ const NewTabPage = React.createClass({
       this.setState({renderedOnce: true});
     }
   },
+  _hasWeightedHighligts() {
+    return this.props.WeightedHighlights.weightedHighlights && this.props.WeightedHighlights.rows.length;
```

Comment:
if `WeightedHighlights` is a reducer, you are enforcing that it always has `rows`, so this check shouldn't be necessary